### ISSUE TITLE
Missing module

### ISSUE THREAD

**Issue by paSPteam** [MicrosoftDocs/OfficeDocs-SharePoint#120]
How do we install the Microsoft.SharePoint.MigrationTool.PowerShell module?

The import fails and so too do the following commands. 

````
PS C:\WINDOWS\system32> Import-Module Microsoft.SharePoint.MigrationTool.PowerShell
````
````
Import-Module : The specified module 'Microsoft.SharePoint.MigrationTool.PowerShell' was not loaded because no valid module file was found in any module directory.
At line:1 char:1
+ Import-Module Microsoft.SharePoint.MigrationTool.PowerShell
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (Microsoft.Share...Tool.PowerShell:String) [Import-Module], FileNotFoundException
    + FullyQualifiedErrorId : Modules_ModuleNotFound,Microsoft.PowerShell.Commands.ImportModuleCommand
````
---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 83229fe4-a0fc-5d45-1e32-2cf9bee9374b
* Version Independent ID: ea237f69-8d65-e3c6-1c77-5f3fdfb73d34
* Content: [Using PowerShell cmdlets to migrate to SharePoint Online](https://docs.microsoft.com/en-us/sharepointmigration/overview-SPMT-PS-cmdlets)
* Content Source: [migration/overview-SPMT-PS-cmdlets.md](https://github.com/MicrosoftDocs/OfficeDocs-SharePoint/blob/live/migration/overview-SPMT-PS-cmdlets.md)
* Product: **sharepoint-server-itpro**
* GitHub Login: @JoanneHendrickson
* Microsoft Alias: **jhendr**

**Comment by wenhua-helen** [comment]
PowerShell migration cmdlets are not released yet. Please stay tuned. 
This article is for coming PowerShell migration release.

**Comment by paSPteam** [comment]
[holds breath :no_mouth:]

**Comment by kenwith** [comment]
@paSPteam - LOL on that emoji. Thanks for taking the time to provide feedback and thanks to @wenhua-helen on the info!

**Comment by potatoqualitee** [comment]
I tried my hardest but still could not find how to download it. Just docs.

**Comment by potatoqualitee** [comment]
Thank you @wenhua-helen! I did indeed find it there after I installed the GUI. Is it possible for your team to make the module available in the [powershellgallery](https://www.powershellgallery.com) as well?

Unfortunately, I don't have any feedback to offer regarding company plans. I'm a PowerShell developer and was curious about the toolset's approach to authentication and which DLLs are being used.

**Comment by wenhua-helen** [comment]
Currently Migration cmdlets are in open beta, we are collecting feedback. Making it a module and publish it in powershellgallery is in our backlog, we will prioritize it with other features. by Now, No ETA yet. 
Please find more information on migration tool's authentication support with below link. https://docs.microsoft.com/en-us/sharepointmigration/how-to-use-the-sharepoint-migration-tool

**Comment by kenwith** [comment]
@potatoqualitee - thanks for this info and I saw Helen already added some info in the docs based on your feedback. I merged into cmdlet docs. Thanks for taking the time on this! Much appreciated!

**Comment by avi081265** [comment]
Is there any way to exclude particular folder using PowerShell cmdlet while migrating from FIle Server to OneDrive for Business?

**Comment by AndreaBarr** [comment]
@paSPteam Thank you for this feedback

If this article doesn't have the PowerShell cmdlet but should help some on the migration preping/excluding files part. https://docs.microsoft.com/en-us/sharepointmigration/create-a-user-mapping-file-for-data-content-migration 

 If this doens't help you enough we want to make sure that you get assistance promply so please open a Support ticket. 

Here is a quick video on opening a ticket: https://www.youtube.com/watch?v=puuOy0sEg1Y 
 
If you don't mind please keep us posted here on the resolution and I really want to get this information you are discovering back into the docs.

**Comment by mikebrunak** [comment]
> Currently Migration cmdlets are in open beta, we are collecting feedback. Making it a module and publish it in powershellgallery is in our backlog, we will prioritize it with other features. by Now, No ETA yet.
> Please find more information on migration tool's authentication support with below link. https://docs.microsoft.com/en-us/sharepointmigration/how-to-use-the-sharepoint-migration-tool

Hello - I see that the commandlets were made available on November 6, and I have downloaded the SharePoint migration tool version 2.1.101.6, but I still cannot find the commandlets. I've tried to install the module but I'm getting"module not found" errors. Is there a location I can download the module?

**Comment by skahlam** [comment]
I am having the same issue.  When will the PS module be available or fixed to import?

**Comment by helenhwl** [comment]
This is the link of SPMT download page.
http://spmtreleasescus.blob.core.windows.net/install/default.htm. looks like the hyperlink is removed after I pasted the messages above.

**Comment by potatoqualitee** [comment]
Consider placing your PowerShell module in the PowerShell Gallery and marking it as pre-release. IThe Gallery is an ideal place to get people aware of your module, trying it out, and providing feedback. And it aligns with what Microsoft suggests and what PowerShell users expect.

**Comment by skahlam** [comment]
I have followed your instructions exactly and I had already performed this before without success.
This is the error I get when I run the help command:

get-help : Get-Help could not find get-spmtmigration in a help file in this session. To download updated help topics type: "Update-Help". To get help online, search for the help topic in the TechNet
library at https:/go.microsoft.com/fwlink/?LinkID=107116.
At line:1 char:1
+ get-help get-spmtmigration
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (:) [Get-Help], HelpNotFoundException
    + FullyQualifiedErrorId : HelpNotFound,Microsoft.PowerShell.Commands.GetHelpCommand



Cheers!


 
Stephanie Kahlam | Consultant
itgroove Professional Services  |  
www.itgroove.net
Office Main: 250.220.4575

**Comment by wenhua-helen** [comment]
Hi, skahlam,
Let us have a skype meeting to trouble shooting the issue. I am in China. Could you please let me know what time is suitable for you to have a skype meeting?


Cheers
Helen

**Comment by helenhwl** [comment]
@potatoqualitee  Thanks for your feedback. Releasing the migration PowerShell module in the PowerShell Gallery is in our backlog, but no ETA yet.

**Comment by wenhua-helen** [comment]
Is SPMT installed successfully? Did you see the logs in page?
if yes, Please share the front end logs  under “C:\Users\%user name%\AppData\Roaming \Microsoft\MigrationTool\FrontEnd.



Cheers
Helen

**Comment by dplotnikov** [comment]
Hey guys, 

Just want to follow-up with you on this and check if you still have any issues with it. First of all, please let me know if you have any content in this folder `%userprofile%\Documents\WindowsPowerShell\Modules`

**Comment by dplotnikov** [comment]
While I'm waiting for your answer, please check my blog post [here](https://dplotnikov.wordpress.com/2018/10/31/sharepoint-migration-tool-v2-is-not-able-to-run-under-administrator/) maybe you have the same issue if this folder is empty `%userprofile%\Documents\WindowsPowerShell\Modules`

**Comment by dplotnikov** [comment]
Just want to make an update here, I've made a PR to the main repo with a clarification that `Import-Module Microsoft.SharePoint.MigrationTool.PowerShell` command should be executed from the DLL location and not from a random place.

**Comment by kenwith** [comment]
This makes sense and I merged it. I think we can close out this issue. If anyone else still having trouble please re-open and we can adjust docs as needed.

**Comment by skahlam** [comment]
I have just downloaded the latest spmt and I am still having the same issue when trying to import-module from the directory where the DLL resides.

**Comment by kenwith** [comment]
@skahlam - OK I re-opened the issue. I am not sure if this is a documentation issue or a product issue. I know @wenhua-helen is the PM from the product team focused on this and @JoanneHendrickson is the doc owner. @scanum --- I would suggest reaching out to @JoanneHendrickson on this one as I know she keeps a close eye on the SPMT content.

**Comment by JoanneHendrickson** [comment]
Skahlam:  Please see this article and see if the resolves your issue.   https://docs.microsoft.com/en-us/sharepointmigration/spmt-install-issues , specifically this section.

**Download the tool instead**

If you still are experiencing unexplained errors, try downloading the SPMT tool to your local computer.
1. Download zip package to your local computer: Download SPMT.zip package.
2. Right-click the zip file, and click properties.
3. On the General tab, select Unblock and then click OK.
4. Open the zip file and extract all files.
5. Run the build.application file.

**Comment by reinoutD** [comment]
I know this conversation is closed, however i had the same problem, and for me the problem appeared to come up by Managed Folder backup. This created two Documents folders. The dll's were stored in the wrong one by the program. After i copied the WindowsPowershell module to the other Documents folder, everything seems to work!

**Comment by potatoqualitee** [comment]
Has this been published to the PowerShell Gallery yet?

**Comment by reinoutD** [comment]
I dont know

**Comment by soyalejolopez** [comment]
Expanding a bit on @reinoutD 's comment since that helped me. Maybe this helps others. If you have KFM setup on your machine and you're redirecting your My Documents folder, import-module won't find the SPMT module. In this case you can: 

- copy the "C:\Users\<username>\Documents\WindowsPowerShell\Modules\Microsoft.SharePoint.MigrationTool.PowerShell" to "C:\Users\<username>\OneDrive - Contoso\Documents\WindowsPowerShell\Modules" then proceed with Import-Module Microsoft.SharePoint.MigrationTool.PowerShell
OR
- specify the full path to the module, for example: import-module "C:\Users\<username>\Documents\WindowsPowerShell\Modules\Microsoft.SharePoint.MigrationTool.PowerShell"

*Note that with the first option, you may not get updates and will have to copy the folder with new updates so that you have the latest DLLs and psd1. The second option will go directly to the directory where the module is installed by SPMT so will catch any updates. This would be the preferred approach, but either should work. At least until we get the module in the PowerShell Gallery. 

@JoanneHendrickson we may want to add similar guidance to our [SPMT powershell page](https://docs.microsoft.com/en-us/sharepointmigration/overview-spmt-ps-cmdlets).
